Y0/y0_euler.py

```diff
@@ -0,0 +1,294 @@
+"""mirgecom driver for the Y0 demonstration.
+
+Note: this example requires a *scaled* version of the Y0
+grid. A working grid example is located here:
+github.com:/illinois-ceesd/data@y0scaled
```

Comment:
This is odd to me. Grids can be scaled in code. Why do we need a separate version of the grid to scale it?

Reply:
I suspected I could scale it in the code - just couldn't figure out how to do it effectively and update all the required grid data structures in my shotgunned effort to get a flow in the Y0 geometry.

My conundrum revolved around the fact that all the current code we have accesses grids only through the eager grudge discretization, e.g. grabbing the nodal coordinates with array_context.  I simply worried that changing the coordinates "out of from under" the discretization infrastructure just wouldn't be a good idea.

Reply:
Yep, "changing out from under" stuff is asking for trouble. The better way is typically "throw the old one away and make a new one". To scale a mesh, there is https://documen.tician.de/meshmode/mesh.html#meshmode.mesh.processing.affine_map.

Reply:
(And if you look at the [source for that](https://github.com/inducer/meshmode/blob/3fe19121f90e96aad2ba7cb30e791cb6ce86de94/meshmode/mesh/processing.py#L1022-L1067), it's not a complicated thing.